server/src/main/java/org/elasticsearch/index/IndexSortConfig.java

```diff
@@ -175,6 +176,9 @@ public Sort buildIndexSort(Function<String, MappedFieldType> fieldTypeLookup,
             if (ft == null) {
                 throw new IllegalArgumentException("unknown index sort field:[" + sortSpec.field + "]");
             }
+            if (Objects.equals(ft.name(), sortSpec.field) == false) {
+                throw new IllegalArgumentException("Cannot use alias [" + sortSpec.field + "] as an index sort field");
```

Comment:
looks like this solution will work also for field aliases defined as runtime fields, once we get your other PR in?

Reply:
Yes, it should cover both situations